### PR TITLE
Part of #9473: Create Vehicle::UpdateMeasurements

### PR DESCRIPTION
**Pull request by hdpoliveira**
Converted from vehicle_update_measurements

### CODE REVIEW

File: src/openrct2/ride/Vehicle.cpp [hunk 1]
```diff
@@ -1665,157 +1665,152 @@ static void vehicle_update_measurements(Vehicle* vehicle)
...
+                curRide->testing_flags &= ~RIDE_TESTING_DROP_UP;
 
-                int16_t z = vehicle->z / 8 - ride->start_drop_height;
-                if (z < 0)
+                int16_t curZ = z / 8 - curRide->start_drop_height;
+                if (curZ < 0)
                 {
-                    z = abs(z);
-                    if (z > ride->highest_drop_height)
+                    curZ = abs(z);
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.cpp:1758]: Shouldn't it be:
```suggestion
                    curZ = abs(curZ);
```

File: src/openrct2/ride/Vehicle.cpp [hunk 2]
```diff
@@ -1665,157 +1665,152 @@ static void vehicle_update_measurements(Vehicle* vehicle)
...
+                curRide->testing_flags &= ~RIDE_TESTING_DROP_DOWN;
 
-                int16_t z = vehicle->z / 8 - ride->start_drop_height;
-                if (z < 0)
+                int16_t curZ = z / 8 - curRide->start_drop_height;
+                if (curZ < 0)
                 {
-                    z = abs(z);
-                    if (z > ride->highest_drop_height)
+                    curZ = abs(z);
```
Comment (tupaschoal) [inline src/openrct2/ride/Vehicle.cpp:1726]: Shouldn't it be:
```suggestion
                    curZ = abs(curZ);
```

File: src/openrct2/ride/Vehicle.cpp [hunk 3]
```diff
@@ -1511,149 +1511,149 @@ static bool vehicle_open_restraints(Vehicle* vehicle)
...
-        ride->CurTestTrackLocation = curTrackLoc;
+        curRide->CurTestTrackLocation = curTrackLoc;
 
-        if (ride_get_entrance_location(ride, ride->current_test_station).isNull())
+        if (ride_get_entrance_location(curRide, curRide->current_test_station).isNull())
             return;
 
-        uint16_t track_elem_type = vehicle->track_type / 4;
-        if (track_elem_type == TRACK_ELEM_POWERED_LIFT || (vehicle->update_flags & VEHICLE_UPDATE_FLAG_ON_LIFT_HILL))
+        uint16_t track_elem_type = track_type / 4;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1595]: change to `camelCase`

File: src/openrct2/ride/Vehicle.cpp [hunk 4]
```diff
@@ -1511,149 +1511,149 @@ static bool vehicle_open_restraints(Vehicle* vehicle)
...
-                if (vehicle->velocity >= 0xB0000)
+                if (velocity >= 0xB0000)
                 {
-                    ride->special_track_elements |= RIDE_ELEMENT_TUNNEL_SPLASH_OR_RAPIDS;
+                    curRide->special_track_elements |= RIDE_ELEMENT_TUNNEL_SPLASH_OR_RAPIDS;
                 }
         }
 
         // ax
         uint16_t track_flags = TrackFlags[track_elem_type];
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1641]: update to `camelCase`. Remove the `ax` comment it is no longer required.

File: src/openrct2/ride/Vehicle.cpp [hunk 5]
```diff
@@ -1511,149 +1511,149 @@ static bool vehicle_open_restraints(Vehicle* vehicle)
...
 
-        uint32_t testing_flags = ride->testing_flags;
+        uint32_t testing_flags = curRide->testing_flags;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1643]: update to camelCase

File: src/openrct2/ride/Vehicle.cpp [hunk 6]
```diff
@@ -1511,149 +1511,149 @@ static bool vehicle_open_restraints(Vehicle* vehicle)
...
-            ride->turn_count_default += 0x800;
+            curRide->turn_count_default += 0x800;
         }
         else if (testing_flags & RIDE_TESTING_TURN_RIGHT || testing_flags & RIDE_TESTING_TURN_LEFT)
         {
-            ride->testing_flags &= ~(
+            curRide->testing_flags &= ~(
                 RIDE_TESTING_TURN_LEFT | RIDE_TESTING_TURN_RIGHT | RIDE_TESTING_TURN_BANKED | RIDE_TESTING_TURN_SLOPED);
 
             uint8_t turn_type = 1;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1659]: updatae to camelCase

File: src/openrct2/ride/Vehicle.cpp [hunk 7]
```diff
@@ -1665,157 +1665,152 @@ static void vehicle_update_measurements(Vehicle* vehicle)
...
         {
-            if (vehicle->velocity < 0 || !(track_flags & TRACK_ELEM_FLAG_DOWN))
+            if (velocity < 0 || !(track_flags & TRACK_ELEM_FLAG_DOWN))
             {
-                ride->testing_flags &= ~RIDE_TESTING_DROP_DOWN;
+                curRide->testing_flags &= ~RIDE_TESTING_DROP_DOWN;
 
-                int16_t z = vehicle->z / 8 - ride->start_drop_height;
-                if (z < 0)
+                int16_t curZ = z / 8 - curRide->start_drop_height;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1723]: replace 8 with `COORDS_Z_STEP`

File: src/openrct2/ride/Vehicle.cpp [hunk 8]
```diff
@@ -1665,157 +1665,152 @@ static void vehicle_update_measurements(Vehicle* vehicle)
...
+            uint8_t drops = curRide->drops & 0x3F;
             if (drops != 0x3F)
                 drops++;
-            ride->drops &= ~0x3F;
-            ride->drops |= drops;
+            curRide->drops &= ~0x3F;
+            curRide->drops |= drops;
 
-            ride->start_drop_height = vehicle->z / 8;
+            curRide->start_drop_height = z / 8;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1745]: replace 8 with `COORDS_Z_STEP `

File: src/openrct2/ride/Vehicle.cpp [hunk 9]
```diff
@@ -1665,157 +1665,152 @@ static void vehicle_update_measurements(Vehicle* vehicle)
...
         {
-            if (vehicle->velocity > 0 || !(track_flags & TRACK_ELEM_FLAG_UP))
+            if (velocity > 0 || !(track_flags & TRACK_ELEM_FLAG_UP))
             {
-                ride->testing_flags &= ~RIDE_TESTING_DROP_UP;
+                curRide->testing_flags &= ~RIDE_TESTING_DROP_UP;
 
-                int16_t z = vehicle->z / 8 - ride->start_drop_height;
-                if (z < 0)
+                int16_t curZ = z / 8 - curRide->start_drop_height;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1755]: `COORDS_Z_STEP `

File: src/openrct2/ride/Vehicle.cpp [hunk 10]
```diff
@@ -1862,37 +1857,37 @@ static void vehicle_update_measurements(Vehicle* vehicle)
 
         if (!cover_found)
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1858]: Can you update the variables to use camelCase in the above.

File: src/openrct2/ride/Vehicle.cpp [hunk 11]
```diff
@@ -1862,37 +1857,37 @@ static void vehicle_update_measurements(Vehicle* vehicle)
...
     }
 
-    if (!(ride->testing_flags & RIDE_TESTING_SHELTERED))
+    if (!(curRide->testing_flags & RIDE_TESTING_SHELTERED))
     {
-        ride->testing_flags |= RIDE_TESTING_SHELTERED;
+        curRide->testing_flags |= RIDE_TESTING_SHELTERED;
 
-        uint8_t num_sheltered_sections = ride->num_sheltered_sections & 0x1F;
+        uint8_t num_sheltered_sections = curRide->num_sheltered_sections & 0x1F;
```
Comment (duncanspumpkin) [inline src/openrct2/ride/Vehicle.cpp:1869]: camelCase

**Pull request merged**